### PR TITLE
docs - update caching

### PR DESCRIPTION
**Pull request by nllho**
Add notes about caching location. Useful for people who need to adhere to data residency laws.

### CODE REVIEW

File: docs/data-modeling/models.md [hunk 1]
```diff
@@ -166,6 +166,12 @@ To set up model caching:
 1. [Enable model caching in Metabase](#enable-model-caching-in-metabase).
 2. [Create a schema to store cached models](#create-a-schema-to-store-cached-models).
 
+### Model caching location
+
+If you're self-hosting Metabase, cached models will be saved to your [application database](../installation-and-operation/configuring-application-database.md).
```
Comment (paoliniluis) [inline docs/data-modeling/models.md:171]: Cached models are saved in the data warehouse, not on the app db :)

File: docs/data-modeling/models.md [hunk 2]
```diff
@@ -166,6 +166,12 @@ To set up model caching:
...
+
+If you're using Metabase Cloud, cached models will be saved to Metabase's servers in the United States.
```
Comment (paoliniluis) [inline docs/data-modeling/models.md:173]: same as above, it's saved on customer's data warehouse in a special schema we create

**Review by dragonsahead — CHANGES_REQUESTED** [review]
Just a few changes

**Comment by nllho** [comment]
Should have read it more carefully -- the model caching section already mentions the schema 😅

**Review by dragonsahead — APPROVED** [review]

**Pull request merged**